hydromt_wflow/wflow.py

```diff
@@ -1606,11 +1631,175 @@ def write_states(self):
         # raise NotImplementedError()
 
     def read_results(self):
-        """Read results at <root/?/> and parse to dict of xr.DataArray"""
+        """Read results at <root/?/> and parse to dict of xr.DataArray/xr.Dataset"""
         if not self._write:
             # start fresh in read-only mode
             self._results = dict()
-        # raise NotImplementedError()
+
+        # Read gridded netcdf (output section)
+        nc_fn = self.get_config("output.path", abs_path=True)
+        if nc_fn is not None and isfile(nc_fn):
+            self.logger.info(f"Read results from {nc_fn}")
+            ds = xr.open_dataset(nc_fn, chunks={"time": 30})
+            # TODO ? align coords names and values of results nc with staticmaps
+            self.set_results(ds, name="output")
+
+        # Read scalar netcdf (netcdf section)
+        ncs_fn = self.get_config("netcdf.path", abs_path=True)
+        if ncs_fn is not None and isfile(ncs_fn):
+            self.logger.info(f"Read results from {ncs_fn}")
+            ds = xr.open_dataset(ncs_fn, chunks={"time": 30})
+            self.set_results(ds, name="netcdf")
+
+        # Read csv timeseries (csv section)
+        csv_fn = self.get_config("csv.path", abs_path=True)
+        if csv_fn is not None and isfile(csv_fn):
+            # Count items by csv.column
+            count = 1
+            # Loop over csv.column
+            for col in self.get_config("csv.column"):
+                header = col["header"]
+                # Column based on map
+                if "map" in col.keys():
+                    # Read the corresponding map and derive the different locations
+                    # The centroid of the geometry is used as coordinates for the timeseries
+                    map_name = self.get_config(f"input.{col['map']}")
+                    da = self.staticmaps[map_name]
+                    gdf = da.raster.vectorize()
+                    gdf.geometry = gdf.geometry.centroid
```

Comment:
I suggest to use the basin outlet or otherwise representative_point as the centroid might be located outside the basin.
https://geopandas.readthedocs.io/en/latest/docs/reference/api/geopandas.GeoSeries.representative_point.html

Reply:
Good suggestion to use the basin outlet! I'll add this in the issue.